### PR TITLE
#171167298 Two factor authentication

### PR DESCRIPTION
**Pull request by gildniy**
#### What does this PR do?
Adds an extra layor of security upon user login (2 Factor Authentication)
#### Description of Task to be completed?
- adds a settings page where user can set method of 2FA to use or disable it
- adds a 2fa page after login when the user has activated one of the methods of 2fa (using sms or authenticator app like Google Authenticator)
#### How should this be manually tested?
- Clone the branch
- Run `npm run watch` to start the project
- Log in to the application and navigate to `settings` page
- Go to `Two factor authentication` link on the sidebar
- And activate one of the methods
- Logout and login, the user should be redirected to the page to validate the token
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#171167298](https://www.pivotaltracker.com/story/show/171206721)
#### Screenshots (if appropriate)

1. Setup first page (Setup)
image

2. Setup scan page (Verification)
image

3. (Login FA SMS)
image

4. (Login FA Authenticator)
image

#### Questions:
N/A

### CODE REVIEW

File: src/components/profile/EditProfile.js [hunk 1]
```diff
@@ -36,6 +36,10 @@ class EditProfile extends Component {
 		});
 	}
 
+	componentWillUnmount() {
+		this.setState();
+	}
+
```
Comment (b0nbon1) [inline src/components/profile/EditProfile.js]: I think you should remove this one. Because it doesn't do anything

File: src/helpers/UseChat.js [hunk 2]
```diff
@@ -12,9 +12,16 @@ const useChat = () => {
 	const dispatch = useDispatch();
 	useEffect(() => {
 		const token = getToken();
-		socketRef.current = socketClient(process.env.API_URL, {
+
+		const connectionOptions = {
+			'force new connection': true,
+			reconnectionAttempts: 'Infinity',
+			timeout: 10000,
+			// transports: ['websocket'],
```
Comment (b0nbon1) [inline src/helpers/UseChat.js]: remove commented line

File: src/components/profile/EditProfile.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gildniy) [reply src/components/profile/EditProfile.js]: Removed.

File: src/store/actions/types.js [hunk 3]
```diff
@@ -40,7 +40,6 @@ export const REQUEST_SEARCH_ERROR = 'REQUEST_SEARCH_ERROR';
 export const SET_LIST = 'SET_LIST';
 export const IS_REQUEST_SEARCHING = 'IS_REQUEST_SEARCHING';
 export const DELETE_COMMENT_SUCCESS = 'DELETE_COMMENT_SUCCESS';
-export const BOOK_ACCOMMODATION_SUCCESS = 'BOOK_ACCOMMODATION_SUCCESS';
```
[the remark this answers is not in the document]
Comment (gildniy) [reply src/store/actions/types.js]: That variable was not used anywhere that's why I removed it.

File: src/components/UserAccount.js [hunk 4]
```diff
@@ -61,17 +58,26 @@ const UserAccount = () => {
...
-						</Link>
-					),
-				)}
+				{/*{[...accountLinks.general, ...(role && accountLinks[role])].map(*/}
+				{/*	(item, idx) => (*/}
+				{/*		<Link key={idx} className='dropdown-item' to={item.linkRoute}>*/}
+				{/*			{item.linkText}*/}
+				{/*		</Link>*/}
+				{/*	),*/}
+				{/*)}*/}
```
Comment (RaymondMwaura) [inline src/components/UserAccount.js]: I think you can remove this commented code.

**Pull request merged**